netdata-installer.sh

```diff
@@ -775,24 +781,34 @@ install_go() {
 		for index in "${ARCH_MAP[@]}" ; do
 			KEY="${index%%::*}"
 			VALUE="${index##*::}"
-			if [ "$KEY" == "$ARCH" ]; then
+			if [ "$KEY" = "$ARCH" ]; then
 				ARCH="${VALUE}"
 				break
 			fi
 		done
 		tmp=$(mktemp -d /tmp/netdata-go-XXXXXX)
-		GO_PACKAGE_BASENAME="go.d.plugin-$GO_PACKAGE_VERSION.$OS-$ARCH"
+		GO_PACKAGE_BASENAME="go.d.plugin-${GO_PACKAGE_VERSION}.${OS}-${ARCH}"
 
-		download "https://github.com/netdata/go.d.plugin/releases/download/$GO_PACKAGE_VERSION/$GO_PACKAGE_BASENAME" "${tmp}/$GO_PACKAGE_BASENAME"
+		download_go "https://github.com/netdata/go.d.plugin/releases/download/${GO_PACKAGE_VERSION}/${GO_PACKAGE_BASENAME}" "${tmp}/${GO_PACKAGE_BASENAME}"
+
+		download_go "https://github.com/netdata/go.d.plugin/releases/download/${GO_PACKAGE_VERSION}/config.tar.gz" "${tmp}/config.tar.gz"
+		if [ ! -f "${tmp}/${GO_PACKAGE_BASENAME}" ] || [ ! -f "${tmp}/config.tar.gz" ]; then
+			run_failed "go.d plugin download failed, go.d plugin will not be available"
+			return 0
+		fi
 
-		download "https://github.com/netdata/go.d.plugin/releases/download/$GO_PACKAGE_VERSION/config.tar.gz" "${tmp}/config.tar.gz"
 		grep "${GO_PACKAGE_BASENAME}\$" "${INSTALLER_DIR}/packaging/go.d.checksums" > "${tmp}/sha256sums.txt" 2>/dev/null
 		grep "config.tar.gz" "${INSTALLER_DIR}/packaging/go.d.checksums" >> "${tmp}/sha256sums.txt" 2>/dev/null
 
 		# Checksum validation
 		if ! (cd "${tmp}" && safe_sha256sum -c "sha256sums.txt"); then
+
+			echo >&2
+			echo >&2 "go.d plugin checksum validation failure."
+			echo >&2 "Either check the error or consider disabling it by issuing '--disable-go' in the installer"
+			echo >&2
+
 			run_failed "go.d.plugin package files checksum validation failed."
```

Comment:
Not sure if this is needed any more. Perhaps it should be combined with the above. Right now it shows:

```
[/home/christopher/CLionProjects/netdata]# curl -sSL --connect-timeout 10 --retry 3 https://github.com/netdata/go.d.plugin/releases/download/v0.6.0/config.tar.gz
curl: (6) Could not resolve host: github.com
FAILED

go.d.plugin-v0.6.0.linux-amd64: FAILED
config.tar.gz: FAILED
sha256sum: WARNING: 2 computed checksums did NOT match

go.d plugin checksum validation failure.
Either check the error or consider disabling it by issuing '--disable-go' in the installer

FAILED  go.d.plugin package files checksum validation failed.

[/home/christopher/CLionProjects/netdata]# rm -rf /usr/lib/netdata/conf.d/go.d
OK

[/home/christopher/CLionProjects/netdata]# rm -rf /usr/lib/netdata/conf.d/go.d.conf
OK

[/home/christopher/CLionProjects/netdata]# tar -xf /tmp/netdata-go-3c7sL3/config.tar.gz -C /usr/lib/netdata/conf.d/
tar: This does not look like a tar archive

gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
FAILED
```

The second is also problematic